### PR TITLE
Added glusterFS, changed workdir for mesos-slave, marathon-port

### PR DESCRIPTION
**Pull request by dvictor1**

### CODE REVIEW

File: roles/common/tasks/Debian.yml [hunk 1]
```diff
@@ -9,3 +12,11 @@
     - unzip
     - ca-certificates
     - software-properties-common
+
```
Comment (jonas) [inline roles/common/tasks/Debian.yml]: @dvictor1 I suggest we move this to a separate `glusterfs` role.

File: roles/common/handlers/main.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+---
+- name: Start GlusterFS
+  service: name=glusterfs-server state=started enabled=true
```
Comment (jonas) [inline roles/common/handlers/main.yml]: Handlers is for running tasks when something changes, and the main use case is for restarting services when a setting changes. So I suggest this is moved to the GlusterFS role and changed to restart the server

File: roles/common/tasks/Debian.yml [hunk 3]
```diff
@@ -9,3 +12,11 @@
...
+- name: Install glusterfs packages
+  apt: name={{item}} state=installed force=yes
+  with_items:
+    - glusterfs-client
+    - glusterfs-server
+  when: "'glusterfs_nodes' in group_names"
+  notify: Start GlusterFS
```
Comment (jonas) [inline roles/common/tasks/Debian.yml]: Here we would then the notification to "restart" GlusterFS.

File: roles/glusterfs/handlers/main.yml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+---
+- name: Restart mesos services
+  service: name={{item}} state=restarted
+  with_items:
```
Comment (jonas) [inline roles/glusterfs/handlers/main.yml:4]: Why must all the services be restarted?

File: roles/glusterfs/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,26 @@
+---
+- name: Set gluster fs facts
+  set_fact:
+    glusterfs_quorum:  "{{ (groups['glusterfs_nodes']) | count | int }}"
```
Comment (jonas) [inline roles/glusterfs/tasks/main.yml]: Should this have the `+ 0.5 | ceil` stuff you added for zookeeper?

File: roles/mesos-master/templates/etc-init-marathon.conf.j2 [hunk 6]
```diff
@@ -12,4 +12,5 @@ script
   --http_credentials "{{ mesos_http_credentials }}" \
   --mesos_authentication_principal "{{ mesos_authentication_principal }}" \
   --mesos_authentication_secret_file "{{ mesos_authentication_secret_file }}"
```
Comment (jonas) [inline roles/mesos-master/templates/etc-init-marathon.conf.j2]: Missing an ending `\`

File: roles/mesos-master/templates/etc-init-marathon.conf.j2 [hunk 7]
```diff
@@ -12,4 +12,5 @@ script
...
+  --http_port {{marathon_port}}
```
Comment (jonas) [inline roles/mesos-master/templates/etc-init-marathon.conf.j2:15]: Why does this have to be passed when it is the default?

File: vagrant/hosts [hunk 8]
```diff
@@ -5,6 +5,7 @@
 [all:vars]
 ansible_ssh_user=vagrant
 ansible_dns=mesos.dev
+marathon_port=8080
```
Comment (jonas) [inline vagrant/hosts:8]: Since this is unlikely to be changed I suggest to define it in `roles/mesos/defaults/main.yml`

File: vagrant/hosts [hunk 9]
```diff
@@ -23,3 +24,7 @@ ansible_dns=mesos.dev
 
 [marathon-lb]
 10.141.141.10
+
+[glusterfs_nodes]
```
Comment (jonas) [inline vagrant/hosts]: I suggest to use `_` or `-` consistently. In this case however, I think `_nodes` can be removed.

File: roles/mesos-agent/tasks/main.yml [hunk 10]
```diff
@@ -12,7 +12,7 @@
     - { file: '/etc/mesos-slave/credential', content: '/etc/mesos-slave/.credentials' }
     - { file: '/etc/mesos-slave/fetcher_cache_size', content: "2GB" }
     - { file: '/etc/mesos-slave/fetcher_cache_dir', content: "/tmp/mesos" }
-    - { file: '/etc/mesos-slave/work_dir', content: "/mnt/mesos" }
+    - { file: '/etc/mesos-slave/work_dir', content: "/log/mesos" }
```
Comment (jonas) [inline roles/mesos-agent/tasks/main.yml:15]: Maybe use a toplevel `/mesos` instead? Unless other apps will log there. Then agent logs will be in `/mesos/slaves`.

File: roles/common/handlers/main.yml (hunk 2 above)
Comment (dvictor1) [inline roles/common/handlers/main.yml]: conditional start here, so I could use notify, I guess. Removed.

File: roles/glusterfs/handlers/main.yml (hunk 4 above)
Comment (dvictor1) [inline roles/glusterfs/handlers/main.yml:4]: not only, but every service, writing to /mnt should be restarted, when /mnt being changed

File: roles/glusterfs/tasks/main.yml (hunk 5 above)
Comment (dvictor1) [inline roles/glusterfs/tasks/main.yml]: this is just a number of gluster hosts. For zookeeper quorum required >50% of servers

File: roles/mesos-agent/tasks/main.yml (hunk 10 above)
Comment (dvictor1) [inline roles/mesos-agent/tasks/main.yml:15]: not sure yet, maybe we gonna need more logs?

File: roles/glusterfs/handlers/main.yml (hunk 4 above)
Comment (jonas) [inline roles/glusterfs/handlers/main.yml:4]: But from my understanding mesos no longer writes to `/mnt`, or am I mistaken?
If I understand correctly we want `/mnt` to only be used by applications. In this case restarting Mesos and Marathon will not restart all applications since docker images would continue to run. Maybe if you also restart docker ;)

Comment (dvictor1) [inline roles/glusterfs/handlers/main.yml:4]: I was thinking about this too :-)

File: roles/glusterfs/tasks/main.yml [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+
+- name: Update APT
+  apt: update_cache=yes cache_valid_time=3600
+
+- name: Install glusterfs packages
+  apt: name={{item}} state=installed force=yes
+  with_items:
+    - glusterfs-client
+    - glusterfs-server
+  when: "'glusterfs' in group_names"
```
Comment (jonas) [inline roles/glusterfs/tasks/main.yml]: This conditional is no longer needed.

File: roles/glusterfs/tasks/main.yml [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+
+- name: Start GlusterFS
+  service: name=glusterfs-server state=started enabled=true
+  when: "'glusterfs' in group_names"
```
Comment (jonas) [inline roles/glusterfs/tasks/main.yml]: This conditional is no longer needed.

File: roles/mesos/tasks/main.yml [hunk 13]
```diff
@@ -22,12 +22,3 @@
   with_items: groups['mesos-dns']
   notify: Regenerate /etc/resolv.conf
 
-# The Mesos package installs both the master and agent. Make the Mesos services
```
Comment (jonas) [inline roles/mesos/tasks/main.yml:25]: Why is this removed? It is needed on Ubuntu and it looks like also RedHat

File: roles/mesos/tasks/RedHat.yml [hunk 14]
```diff
@@ -6,3 +6,21 @@
   yum: name={{item}} state=installed
   with_items:
     - mesos-{{mesos_version}}
+
+# The Mesos package installs both the master and agent. Make the Mesos services
+# match those of the roles.
+
+- name: Check Mesos Master running
```
Comment (jonas) [inline roles/mesos/tasks/RedHat.yml:13]: I think the declarative way of doing this by using `service: name=... state=stopped` is much cleaner. Let's keep the rules in `roles/mesos/tasks/main.yml`

File: roles/glusterfs/tasks/main.yml (hunk 12 above)
Comment (dvictor1) [inline roles/glusterfs/tasks/main.yml]: sure!

File: roles/mesos/tasks/main.yml (hunk 13 above)
Comment (dvictor1) [inline roles/mesos/tasks/main.yml:25]: it's moved into Debian and Redhat branches. I noticed that Redhat works in a different way with services.

File: roles/mesos/tasks/RedHat.yml (hunk 14 above)
Comment (dvictor1) [inline roles/mesos/tasks/RedHat.yml:13]: works for ubuntu, but not so great on Redhat. One of Todo's.

**Comment by jonas** [comment]
:+1:
